### PR TITLE
docs: module variables

### PR DESCRIPTION
**Pull request by LingyuCoder**
## Summary

Update documents of module variables

## Checklist

- [ ] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

File: website/docs/en/api/modules/module-variables.mdx [hunk 1]
```diff
@@ -126,52 +240,371 @@ componentsContext.keys().forEach(fileName => {
 
 `import.meta.webpackContext()` streamlines the process of module importation especially when you have a lot of files to manage. When using it, please avoid matching unnecessary files, as this might lead to significantly increased build time and output size.
 
-### import.meta.webpackHot (webpack-specific)
+### import.meta.webpackHot (Rspack/Webpack-specific)
```
Comment (chenjiahan) [inline website/docs/en/api/modules/module-variables.mdx:243]: How about removing `(Rspack/Webpack-specific)` from the title? It makes the title too long

Comment (LingyuCoder) [reply website/docs/en/api/modules/module-variables.mdx:243]: This is used to distinguish from the standard. Perhaps we can try to use some label components

Comment (LingyuCoder) [reply website/docs/en/api/modules/module-variables.mdx:243]: I will handle this in an other pr

Comment (chenjiahan) [reply website/docs/en/api/modules/module-variables.mdx:243]: How about move to comment:

```md
### import.meta.webpackHot

> This is an Rspack/Webpack specific API.
```

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**